Lib9c/Model/Stat/CharacterStats.cs

```diff
@@ -249,85 +186,67 @@ public CharacterStats SetStats(int level, bool updateImmediate = true)
-                            statMap.HP));
-                    }
-
-                    if (statMap.HasATK)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.ATK, StatModifier.OperationType.Add,
-                            statMap.ATK));
-                    }
-
-                    if (statMap.HasDEF)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.DEF, StatModifier.OperationType.Add,
-                            statMap.DEF));
-                    }
-
-                    if (statMap.HasCRI)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.CRI, StatModifier.OperationType.Add,
-                            statMap.CRI));
-                    }
-
-                    if (statMap.HasHIT)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.HIT, StatModifier.OperationType.Add,
-                            statMap.HIT));
-                    }
-
-                    if (statMap.HasSPD)
+                    foreach (var (statType, value) in statMap.GetStats(true))
                     {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.SPD, StatModifier.OperationType.Add,
-                            statMap.SPD));
+                        var statModifier = new StatModifier(
+                            statType,
+                            StatModifier.OperationType.Add,
+                            value);
+                        _equipmentStatModifiers.Add(statModifier);
                     }
+                }
+            }
 
-                    if (statMap.HasDRV)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.DRV, StatModifier.OperationType.Add,
-                            statMap.DRV));
-                    }
+            if (updateImmediate)
+            {
+                UpdateConsumableStats();
+            }
 
-                    if (statMap.HasDRR)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.DRR, StatModifier.OperationType.Add,
-                            statMap.DRR));
-                    }
+            return this;
+        }
 
-                    if (statMap.HasCDMG)
-                    {
-                        _consumableStatModifiers.Add(new StatModifier(StatType.CDMG, StatModifier.OperationType.Add,
-                            statMap.CDMG));
-                    }
+        /// <summary>
+        /// Set stats based on runes.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="updateImmediate"></param>
+        /// <returns></returns>
+        public CharacterStats SetRunes(IEnumerable<StatModifier> value, bool updateImmediate = true)
+        {
```

Comment:
I wonder if this new specification is backward compatible.

Reply:
It's not. I should fix it when handling state mismatches.